solenoid/records/models.py

```diff
@@ -65,26 +65,38 @@ def create_citation(paper_data):
         Some Elements papers include the citation field in their metadata,
         but some leave it blank; in those cases we need to build our own
         citation text. Stakeholders have indicated that it doesn't matter what
-        citation format we use or whether the citation is complete (as long as
-        it includes author, title, and journal title).
-
+        citation format we use or whether the citation is complete.
         This method assumes that the supplied paper metadata has already been
         validated; it does not perform any validation.
 
+        If any of the citation fields are missing, the following placeholder text
+        will be used in the citation: "<[CITATION FIELD NAME] UNIDENTIFIED>".
+
+        Note: The citation field name will be formatted such that hyphens are removed
+              and all characters are uppercase.
+
+
         APA format is:
             Author, F.M. (Publication year). Article Title. Journal Title,
             Volume(Issue), pp.-pp. doi:XX.XXXXX.
         We don't appear to get page number information, so we'll skip that.
         """
+        citation_fields = {
+            field: paper_data.get(
+                field, f"<{field.replace('-', ' ').upper()} UNIDENTIFIED>"
+            )
+            for field in Fields.CITATION_DATA
+        }
         citation = "{last}, {first_init}. ".format(
-            last=paper_data[Fields.LAST_NAME], first_init=paper_data[Fields.FIRST_NAME][0]
+            last=citation_fields[Fields.LAST_NAME],
+            first_init=citation_fields[Fields.FIRST_NAME][0],
```

Comment:
I think I see the thinking behind `citation_fields` dictionary, where it loops through all fields and if not present in the data, sets a string like `<Field Name UNIDENTIFIED>`.  

But on line 92, what happens if `Fields.FIRST_NAME` was not present in the data?  Wouldn't `citation_fields` have a string value `"<First Name UNIDENTIFIED>"` and therefore this index retrieval of `[0]` would return the first character of that string `"<"`?

Additionally, I think it could get easy to miss that `citation_fields` and `paper_data` are similar, but not identical, but both are used throught the rest of the method.

I'm not sure what an alternative would be offhand, as other methods in this class are using kind of similar logic of looping through `Fields.*` and doing things.  

If that _is_ a possible bug outlined above, but that's the only real blocker, might just be worth figuring out a workaround for that and moving on!

Reply:
Good point! Please see the changes introduced [here](https://github.com/MITLibraries/solenoid/pull/472/commits/33d781306f49dc688828522bca4c0c6d200be68f#diff-f44916c77edb5e870826c4902360a7d7ca503915c9241b2a08eede211639ea39R86-R112).

Reply:
@jonavellecuerdo - I dig it.  This kind of string building, coupled with logic to account for the pre-existing enums and structures, is definitely kind of finicky.  But this new approach, I find it much easier to just scan and understand.